changelog/unreleased/4671

```diff
@@ -0,0 +1,6 @@
+Bugfix: Set authorization header for token request only when necessary
+
+A new field has been added to the token request to indicate whether the authorization header should be applied.
```

Comment:
I'd be more accurate here, like:

```
A new field has been added to the token request to indicate whether the authorization header should be applied in case `client_secret_post` method is supported or not.
```